app/models/concerns/authorizable.rb

```diff
@@ -1,13 +1,52 @@
 module Authorizable
   extend ActiveSupport::Concern
 
-  included do
-    def authorized?(permission)
-      return false if User.current.nil?
-      User.current.can?(permission, self)
+  def check_permissions_after_save
+    authorizer = Authorizer.new(User.current)
+    creation = self.id_changed?
+    name = permission_name(creation ? :create : :edit)
+
+    Foreman::Logging.logger('permissions').debug { "verifying the transaction by permission #{name} for class #{self.class}" }
+    unless authorizer.can?(name, self)
+      org_loc_string = Taxonomy.enabled_taxonomies.map { |tax| _(tax) }.join(_(" or "))
```

Comment:
nitpick, best to add the spaces in the string below rather than in the translated string